File: 2023.2/TEP/ATCODER4/D.py
def count_triples(N, grid):
    count = 0

    # Verifica todas as combinações possíveis de triples de células
    for i in range(N):
        for j in range(N):
            if grid[i][j] == 'o':
                # Verifica todas as células na mesma linha
                for k in range(N):
                    if k != j and grid[i][k] == 'o':
                        # Verifica todas as células na mesma coluna
                        for l in range(N):
                            if l != i and grid[l][j] == 'o':
                                count += 1

    return count // 2  # Divide por 2 para evitar a contagem duplicada

# Leitura de entrada
N = int(input())
grid = [input() for _ in range(N)]

# Chamada da função e impressão do resultado
result = count_triples(N, grid)
print(result)

File: 2023.2/TEP/ATCODER4/D.c
#include <stdio.h>

#define MAX 2010

char m[MAX][MAX];

int solve(char m[MAX][MAX], int N) {
    int count = 0;

    int row_count[MAX] = {0};
    int col_count[MAX] = {0};

    // Contagem de "o" por linha e coluna
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            if (m[i][j] == 'o') {
                row_count[i]++;
                col_count[j]++;
            }
        }
    }

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            if (m[i][j] == 'o') {
                count += (row_count[i] - 1) * (col_count[j] - 1);
            }
        }
    }

    return count;
}

int main() {
    int N;

    scanf("%d", &N);
    for (int i = 0; i < N; i++) {
        scanf("%s", m[i]);
    }

    printf("%d\n", solve(m, N));

    return 0;
}
